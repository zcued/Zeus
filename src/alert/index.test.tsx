import * as React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import Alert from '.'

describe('Alert', () => {
  it('sets data attributes on alert', () => {
    const wrapper = shallow(<Alert data-test="test-id" data-id="12345" />)
    expect(wrapper.prop('data-test')).toBe('test-id')
    expect(wrapper.prop('data-id')).toBe('12345')
  })

  it('sets aria attributes on alert', () => {
    const wrapper = shallow(<Alert aria-describedby="some-label" />)

    expect(wrapper.prop('aria-describedby')).toBe('some-label')
  })

  it('sets role attribute on alert', () => {
    const wrapper = shallow(<Alert role="status" />)
    expect(wrapper.prop('role')).toBe('status')
  })

  it('sets title on alert', () => {
    const wrapper = shallow(<Alert title="status" />)
    expect(wrapper).toMatchSnapshot()
  })
})
