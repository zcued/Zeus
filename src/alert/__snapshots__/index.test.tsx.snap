// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert sets title on alert 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  display: inline-flex;
  align-items: center;
  color: ",
          [Function],
          ";
  flex: none;
  border-radius: 100px;
  font-size: ",
          [Function],
          "px;
  padding: ",
          [Function],
          "px ",
          [Function],
          "px;
  background-color: ",
          [Function],
          ";
  box-sizing: border-box;
  position: relative;
  line-height: 1;

  & > div[data-icon='true'] {
    margin-right: ",
          [Function],
          "px;
  }
",
        ],
      },
      "displayName": "Styled(Alert)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    }
  }
  forwardedRef={null}
  title="status"
/>
`;
